libsigner/src/v0/messages.rs

```diff
@@ -0,0 +1,538 @@
+}
+
+define_u8_enum!(RejectCodeTypePrefix{
+    ValidationFailed = 0,
+    ConnectivityIssues = 1
+});
+
+impl TryFrom<u8> for RejectCodeTypePrefix {
+    type Error = CodecError;
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        Self::from_u8(value).ok_or_else(|| {
+            CodecError::DeserializeError(format!("Unknown reject code type prefix: {value}"))
+        })
+    }
+}
+
+impl From<&RejectCode> for RejectCodeTypePrefix {
+    fn from(reject_code: &RejectCode) -> Self {
+        match reject_code {
+            RejectCode::ValidationFailed(_) => RejectCodeTypePrefix::ValidationFailed,
+            RejectCode::ConnectivityIssues => RejectCodeTypePrefix::ConnectivityIssues,
+        }
+    }
+}
+
+/// This enum is used to supply a `reason_code` for block rejections
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum RejectCode {
+    /// RPC endpoint Validation failed
+    ValidationFailed(ValidateRejectCode),
+    /// The block was rejected due to connectivity issues with the signer
+    ConnectivityIssues,
+}
+
+/// The response that a signer sends back to observing miners
+/// either accepting or rejecting a Nakamoto block with the corresponding reason
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum BlockResponse {
+    /// The Nakamoto block was accepted and therefore signed
+    Accepted((Sha512Trunc256Sum, MessageSignature)),
+    /// The Nakamoto block was rejected and therefore not signed
+    Rejected(BlockRejection),
+}
+
+#[cfg_attr(test, mutants::skip)]
+impl std::fmt::Display for BlockResponse {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            BlockResponse::Accepted(a) => {
+                write!(
+                    f,
+                    "BlockAccepted: signer_sighash = {}, signature = {}",
+                    a.0, a.1
+                )
+            }
+            BlockResponse::Rejected(r) => {
+                write!(
+                    f,
+                    "BlockRejected: signer_sighash = {}, code = {}, reason = {}",
+                    r.reason_code, r.reason, r.signer_signature_hash
+                )
+            }
+        }
+    }
+}
+
+impl BlockResponse {
+    /// Create a new accepted BlockResponse for the provided block signer signature hash and signature
+    pub fn accepted(hash: Sha512Trunc256Sum, sig: MessageSignature) -> Self {
+        Self::Accepted((hash, sig))
+    }
+
+    /// Create a new rejected BlockResponse for the provided block signer signature hash and rejection code
+    pub fn rejected(hash: Sha512Trunc256Sum, reject_code: RejectCode) -> Self {
+        Self::Rejected(BlockRejection::new(hash, reject_code))
+    }
+}
+
+impl StacksMessageCodec for BlockResponse {
+    fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), CodecError> {
+        match self {
+            BlockResponse::Accepted((hash, sig)) => {
+                write_next(fd, &0u8)?;
```

Comment:
Can you have named values here instead of bare u8s for the type prefix?

Reply:
Will do